en/services/gimbal_v2.md

```diff
@@ -0,0 +1,358 @@
+
+In this set-up the gimbal manager can be on the onboard computer.
+For this case all commands for gimbal manager need to be sent to the gimbal manager (in this case the companion computer), and the messages to the gimbal device need to be sent/routed to the autopilot.
+
+<!-- Mermaid graph:
+graph LR
+	ap["Autopilot"]
+	cc["Companion (Gimbal Manager)"]
+	g["Gimbal Device"]
+	gcs["Ground Station"]
+	ap ---|"Gimbal device messages"|g
+	ap ---|"Gimbal device messages"|cc
+	gcs ---|"Gimbal manager messages"|cc
+-->
+
+[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggTFJcblx0YXBbXCJBdXRvcGlsb3RcIl1cblx0Y2NbXCJDb21wYW5pb24gKEdpbWJhbCBNYW5hZ2VyKVwiXVxuXHRnW1wiR2ltYmFsIERldmljZVwiXVxuXHRnY3NbXCJHcm91bmQgU3RhdGlvblwiXVxuXHRhcCAtLS18XCJHaW1iYWwgZGV2aWNlIG1lc3NhZ2VzXCJ8Z1xuXHRhcCAtLS18XCJHaW1iYWwgZGV2aWNlIG1lc3NhZ2VzXCJ8Y2Ncblx0Z2NzIC0tLXxcIkdpbWJhbCBtYW5hZ2VyIG1lc3NhZ2VzXCJ8Y2MiLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggTFJcblx0YXBbXCJBdXRvcGlsb3RcIl1cblx0Y2NbXCJDb21wYW5pb24gKEdpbWJhbCBNYW5hZ2VyKVwiXVxuXHRnW1wiR2ltYmFsIERldmljZVwiXVxuXHRnY3NbXCJHcm91bmQgU3RhdGlvblwiXVxuXHRhcCAtLS18XCJHaW1iYWwgZGV2aWNlIG1lc3NhZ2VzXCJ8Z1xuXHRhcCAtLS18XCJHaW1iYWwgZGV2aWNlIG1lc3NhZ2VzXCJ8Y2Ncblx0Z2NzIC0tLXxcIkdpbWJhbCBtYW5hZ2VyIG1lc3NhZ2VzXCJ8Y2MiLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)
+
+
+### Multiple gimbals
+
+Multiple gimbals per drone are supported.
+
+#### Component IDs
+
+Multiple component IDs are reserved for gimbal devices: `MAV_COMP_ID_GIMBAL`, `MAV_COMP_ID_GIMBAL2`, `MAV_COMP_ID_GIMBAL3`, `MAV_COMP_ID_GIMBAL4`, `MAV_COMP_ID_GIMBAL5`, `MAV_COMP_ID_GIMBAL6`.
+
+The listed component IDs should be used where possible (other ids may be used as long as the [MAV_TYPE](../messages/common.md#MAV_TYPE) is correctly set to [MAV_TYPE_GIMBAL](../messages/common.md#MAV_TYPE_GIMBAL)).
+
+#### Mapping from gimbal managers to gimbal devices
+
+Each gimbal manager needs to announce which gimbal it maps to.
+It is a 1:1 relationship, each gimbal manager is responsible for one gimbal device. However, multiple gimbal managers can be implemented on the same MAVLink component.
+E.g. an autopilot can implement two gimbal managers in order to control two gimbal devices.
+
+#### Addressing of gimbal devices
+
+Gimbal manager commands and messages have a param respective field to indicate the gimbal device component ID that they intend to control. However, the message needs to be sent to the gimbal manager. If all gimbal devices should be controlled this param/field can be set to 0 signalling "all".
+
+## Implementation and messages
+
+### Messages between ground station and gimbal manager
+
+#### Discovery of gimbal manager
+
+A ground station should initially discover all gimbal managers by sending [MAV_CMD_REQUEST_MESSAGE](#MAV_CMD_REQUEST_MESSAGE) to request [GIMBAL_MANAGER_INFORMATION](#GIMBAL_MANAGER_INFORMATION) from any sysid and compid.
```

Comment:
what do you mean by `from any sysid and compid.`.

Is this to a broadcast ID so everyone responds? Or is it that the message should be sent to every system until you find a gimbal manager?

Reply:
Ditto for the device.

Reply:
Yes that's what it is supposed to mean.

Reply:
Fixed. Recheck